internal/pkg/logging/workload.go

```diff
@@ -31,200 +30,277 @@ type logGetter interface {
 	LogEvents(opts cloudwatchlogs.LogEventsOpts) (*cloudwatchlogs.LogEventsOutput, error)
 }
 
-// WorkloadClient retrieves the logs of an Amazon ECS or AppRunner service.
-type WorkloadClient struct {
-	name         string
-	logGroupName string
-	isECS        bool
+type serviceARNGetter interface {
+	ServiceARN(env string) (string, error)
+}
+
+// NewWorkloadLoggerOpts contains fields that initiate workloadLogger struct.
+type NewWorkloadLoggerOpts struct {
+	App  string
+	Env  string
+	Name string
+	Sess *session.Session
+}
+
+// newWorkloadLogger returns a workloadLogger for the svc service under env and app.
+// The logging client is initialized from the given sess session.
+func newWorkloadLogger(opts *NewWorkloadLoggerOpts) *workloadLogger {
+	return &workloadLogger{
+		app:          opts.App,
+		env:          opts.Env,
+		name:         opts.Name,
+		eventsGetter: cloudwatchlogs.New(opts.Sess),
+		w:            log.OutputWriter,
+		now:          time.Now,
+	}
+}
+
+type workloadLogger struct {
+	app  string
+	env  string
+	name string
+
 	eventsGetter logGetter
 	w            io.Writer
-
-	now func() time.Time
+	now          func() time.Time
 }
 
-// WriteLogEventsOpts wraps the parameters to call WriteLogEvents.
-type WriteLogEventsOpts struct {
-	Follow    bool
-	Limit     *int64
-	StartTime *int64
-	EndTime   *int64
-	TaskIDs   []string
-	// OnEvents is a handler that's invoked when logs are retrieved from the service.
-	OnEvents func(w io.Writer, logs []HumanJSONStringer) error
-	// LogStreamLimit is an optional parameter for jobs and tasks to speed up CW queries
-	// involving multiple log streams.
-	LogStreamLimit          int
-	IncludeStateMachineLogs bool
-	ContainerName           string
-}
-
-// NewWorkloadLogsConfig contains fields that initiates WorkloadClient struct.
-type NewWorkloadLogsConfig struct {
-	App         string
-	Env         string
-	Name        string
-	Sess        *session.Session
-	LogGroup    string
-	WkldType    string
-	TaskIDs     []string
-	ConfigStore describe.ConfigStoreSvc
+// WriteLogEvents writes service logs.
+func (s *workloadLogger) writeEventLogs(logEventsOpts cloudwatchlogs.LogEventsOpts, onEvent func(io.Writer, []HumanJSONStringer) error, follow bool) error {
+	for {
+		logEventsOutput, err := s.eventsGetter.LogEvents(logEventsOpts)
+		if err != nil {
+			return fmt.Errorf("get log events for log group %s: %w", logEventsOpts.LogGroup, err)
+		}
+		if err := onEvent(s.w, cwEventsToHumanJSONStringers(logEventsOutput.Events)); err != nil {
+			return err
+		}
+		if !follow {
+			return nil
+		}
+		// for unit test.
+		if logEventsOutput.StreamLastEventTime == nil {
+			return nil
+		}
+		logEventsOpts.StreamLastEventTime = logEventsOutput.StreamLastEventTime
+		time.Sleep(cloudwatchlogs.SleepDuration)
+	}
 }
 
-func (o WriteLogEventsOpts) limit() *int64 {
-	if o.Limit != nil {
-		return o.Limit
+func (s *workloadLogger) ecsLogStreamPrefixes(taskIDs []string, container string) []string {
```

Comment:
nit: i feel like this function doesn't belong on `workloadLogger` - maybe change the signature to something like
```
func ecsLogStreamPrefixes(service, container string, taskIDs []string) []string 
```